packages/flutter_tools/lib/src/usage.dart

```diff
@@ -19,11 +20,14 @@ const String _kFlutterUA = 'UA-67589403-6';
 Usage get flutterUsage => Usage.instance;
 
 class Usage {
-  /// Create a new Usage instance; [versionOverride] is used for testing.
-  Usage({ String settingsName: 'flutter', String versionOverride }) {
+  /// Create a new Usage instance; [versionOverride] and [configDirOverride] are
+  /// used for testing.
+  Usage({ String settingsName: 'flutter', String versionOverride, String configDirOverride}) {
     final FlutterVersion flutterVersion = FlutterVersion.instance;
     final String version = versionOverride ?? flutterVersion.getVersionString(whitelistBranchName: true);
-    _analytics = new AnalyticsIO(_kFlutterUA, settingsName, version);
+    _analytics = new AnalyticsIO(_kFlutterUA, settingsName, version,
+        // ignore: argument_type_not_assignable
```

Comment:
can you add a comment here explaining the ignore?

Reply:
I won't pretend to fully understand the need for this, but will repeat @tvolkert's explanation here when I asked him about this problem.  I can't seem to track down the location of the conditional import in question, which is probably not that surprising considering I'd ordinarily use the analyzer to figure that out...